### PR TITLE
Uses NuGet.Packaging API to provide some base functionality

### PR DESCRIPTION
**Pull request by petertiedemann**
@Scott-Harris The tests are now green on this, but apart from that i have no idea if its working or not :)

I managed to replace some of the basic Package types and IO with the Nuget.Packaging API (strictly speaking it belongs to Nuget v3, but the 2.8.8 API is pretty horrible.

There is more that can be replaced, but to keep things incremental i will probably be looking to merge this soon assuming i get around to test it a bit.

### CODE REVIEW

File: CsMerge/PackagesConfigMerger.cs.orig [hunk 1]
```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
```
Comment (Scott-Harris) [inline CsMerge/PackagesConfigMerger.cs.orig]: Should this file be included?

File: CsMerge/UserResolvers.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
+﻿using System;
```
Comment (Scott-Harris) [inline CsMerge/UserResolvers.cs]: This file isn't in the project. Can we remove it?

File: CsMerge/UserResolvers.cs.orig [hunk 3]
```diff
@@ -0,0 +1,99 @@
+﻿using System;
```
Comment (Scott-Harris) [inline CsMerge/UserResolvers.cs.orig]: Remove?

File: CsMerge/UserResolvers.cs (hunk 2 above)
Comment (petertiedemann) [inline CsMerge/UserResolvers.cs]: Yes, probably snuck in during the rebase.

File: CsMerge/UserResolvers.cs.orig (hunk 3 above)
Comment (petertiedemann) [inline CsMerge/UserResolvers.cs.orig]: Yep, we need to update the .gitignore file :)

File: CsMerge/PackagesConfigMerger.cs.orig (hunk 1 above)
Comment (petertiedemann) [inline CsMerge/PackagesConfigMerger.cs.orig]: No, .gitignore in need of fixing...

**Pull request closed without merging**